### PR TITLE
add pebble version check

### PR DESCRIPTION
**Pull request by poonai**
closes: https://github.com/go-bond/bond/issues/124

### CODE REVIEW

File: bond.go [hunk 1]
```diff
@@ -141,6 +142,43 @@ func Open(dirname string, opts *Options) (DB, error) {
 		opts.PebbleOptions = DefaultPebbleOptions()
 	}
 
+	_, err := os.Stat(dirname)
```
Comment (marino39) [inline bond.go:145]: Please use the path ``path.Join(dirname, "bond")``. We had issues before with storing files along sstables. It's safer to have additional files in subdir.

**Review by marino39 — CHANGES_REQUESTED** [review]
One small comment otherwise it LGTM.

File: bond.go (hunk 1 above)
Comment (marino39) [reply bond.go:145]: Or maybe put sstables in subdir WDYT?

```
bond/pebble
bond/PEBBLE_FORMAT_VERSION
```

File: bond.go [hunk 2]
```diff
@@ -141,6 +142,43 @@ func Open(dirname string, opts *Options) (DB, error) {
 		opts.PebbleOptions = DefaultPebbleOptions()
 	}
 
+	_, err := os.Stat(filepath.Join(dirname, "bond"))
```
Comment (marino39) [inline bond.go:145]: Please consider assigning this to var instead of recalculating paths all the time.

**Review by marino39 — APPROVED** [review]
One more minor comment, otherwise LGTM.

**Comment by pkieltyka** [comment]
very helpful.

what should we do in cases when we want for bond to proceed to open..?

**Comment by poonai** [comment]
@pkieltyka we have migration framework planned for that. 

https://github.com/go-bond/bond/issues/126

**Comment by marino39** [comment]
BTW: @poonai I forgot that the ticket also mentioned implementing the pebble format migration mechanism. Please do that in this ticket too.

File: bond.go [hunk 3]
```diff
@@ -141,6 +142,45 @@ func Open(dirname string, opts *Options) (DB, error) {
...
+		return nil, err
+	}
+	if err != nil && os.IsNotExist(err) {
+		// create dir if db dir didn't exit.
+		if err := os.MkdirAll(bondPath, os.ModePerm); err != nil {
+			return nil, err
+		}
+	}
+
+	pebbelVersionPath := filepath.Join(bondPath, "PEBBLE_FORMAT_VERSION")
```
Comment (marino39) [inline bond.go:157]: Use const for ``bond`` and ``PEBBLE_FORMAT_VERSION``. It just makes any changes easier by just editing the const.

**Review by marino39 — APPROVED** [review]
Please apply one last comment and merge the PR.

**Pull request merged**